### PR TITLE
update kubelinter, controller-runtime and also Go versions

### PR DESCRIPTION
**Pull request by tremes**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,25 +1,139 @@
 module github.com/app-sre/deployment-validation-operator
 
-go 1.16
+go 1.18
```
Comment (tremes) [inline go.mod:3]: This will probably not work, because the image seems to have go 1.17

**Comment by tremes** [comment]
Not sure why the prow is still `{"level":"info","msg":"Tagging openshift/boilerplate:image-v2.3.2 into pipeline:root.","time":"2023-01-26T14:29:06Z"}`

**Comment by tremes** [comment]
I am failing to see why the CI is still using the original ` quay.io/app-sre/boilerplate:image-v2.3.2`. Any ideas please @Ajpantuso ?

**Comment by Ajpantuso** [comment]
> I am failing to see why the CI is still using the original ` quay.io/app-sre/boilerplate:image-v2.3.2`. Any ideas please @Ajpantuso ?  

You need to update https://github.com/app-sre/deployment-validation-operator/blob/master/.ci-operator.yaml as well.

**Comment by tremes** [comment]
@Ajpantuso thanks

**Comment by tremes** [comment]
I still don't understand why the `make validate` target is failing here. @Ajpantuso do you know please?

**Comment by Ajpantuso** [comment]
> I still don't understand why the `make validate` target is failing here. @Ajpantuso do you know please? 

I would follow the update instructions for boilerplate https://github.com/openshift/boilerplate/blob/master/README.md first and perform that update in a single PR before this one. As you notes that will only give you go1.17, but it should avoid the dirty repo errors.

**Comment by tremes** [comment]
OK I created https://github.com/app-sre/deployment-validation-operator/pull/227. Thanks

**Pull request merged**